Source/Classes/Categories/NSURLSessionTask+Timing.m

```diff
@@ -11,111 +11,127 @@
 #import <objc/runtime.h>
 #import <QuartzCore/QuartzCore.h>
 
-static NSString * const kPINURLSessionTaskStateKey = @"state";
+static inline void pin_swizzleSelector(Class theClass, SEL originalSelector, SEL swizzledSelector) {
+    Method originalMethod = class_getInstanceMethod(theClass, originalSelector);
+    Method swizzledMethod = class_getInstanceMethod(theClass, swizzledSelector);
+    method_exchangeImplementations(originalMethod, swizzledMethod);
+}
 
-@interface PINURLSessionTaskObserver : NSObject
+static inline BOOL pin_addMethod(Class theClass, SEL selector, Method method) {
+    return class_addMethod(theClass, selector,  method_getImplementation(method),  method_getTypeEncoding(method));
+}
 
-@property (atomic, assign) CFTimeInterval startTime;
-@property (atomic, assign) CFTimeInterval endTime;
 
-- (instancetype)init NS_UNAVAILABLE;
-- (instancetype)initWithTask:(NSURLSessionTask *)task NS_DESIGNATED_INITIALIZER;
+@interface _PINURLSessionTaskSwizzling : NSObject
 
 @end
 
-@implementation PINURLSessionTaskObserver
+@implementation _PINURLSessionTaskSwizzling
 
-- (instancetype)initWithTask:(NSURLSessionTask *)task
-{
-    if (self = [super init]) {
-        _startTime = 0;
-        _endTime = 0;
-        [task addObserver:self forKeyPath:kPINURLSessionTaskStateKey options:0 context:nil];
++ (void)load {
```

Comment:
I wonder if there's a way to avoid using `load`? This will increase startup time which I like to avoid as much as possible. Could we possibly make this `PIN_setupSessionTaskObserver` and make it a class method with a dispatch_once inside? Then put all the code within a dispatch_once?

Reply:
I also prefer to avoid using of `load`, which is introduced for method swizzling. But I don't know how to update the `PIN_startTime` correctly instead of hooking `resume` method. In order to make method swizzling work correctly, I think `load` method cannot be avoid. 

Do you think it's right if I add a new method call to update `PIN_startTime` manually after call of [task's `resume`](https://github.com/pinterest/PINRemoteImage/blob/master/Source/Classes/PINRemoteImageDownloadQueue.m#L108)?